cilium/network_policy.h

```diff
@@ -344,5 +344,42 @@ class NetworkPolicyMap : public Singleton::Instance,
 };
 using NetworkPolicyMapSharedPtr = std::shared_ptr<const NetworkPolicyMap>;
 
+struct SNIPattern {
+  std::string pattern;
+
+  explicit SNIPattern(const std::string& p) : pattern(absl::AsciiStrToLower(p)) {}
+
+  bool matches(const absl::string_view sni) const {
+    if (pattern.empty() || sni.empty()) {
+      return false;
+    }
+    auto const lower_sni = absl::AsciiStrToLower(sni);
+    // Perform lower case exact match if there is no wildcard prefix
+    if (!pattern.starts_with("*")) {
+      return pattern == lower_sni;
+    }
+
+    // Pattern is "**.<domain>"
+    if (pattern.starts_with("**.")) {
+      return lower_sni.ends_with(pattern.substr(2));
+    }
+
+    // Pattern is "*.<domain>"
+    if (pattern.starts_with("*")) {
```

Comment:
IMO we should not allow patterns without the dot after the asterisk, so this should be `starts_with("*.")` instead.

Reply:
thanks and updated :heavy_check_mark: